src/clj/puppetlabs/trapperkeeper/services/watcher/filesystem_watch_core.clj

```diff
@@ -30,9 +30,11 @@
   [event :- WatchEvent
    watch-path :- Path]
   {:type (get event-type-mappings (.kind event))
-   :path (-> watch-path
-             (.resolve (.context event))
-             fs/file)})
+   :path (if (= StandardWatchEventKinds/OVERFLOW (.kind event))
+           (fs/file (.resolve watch-path "UNKNOWN"))
```

Comment:
We talked about this in standup. This request makes good sense. I've adopted it nearly wholesale.

The only change I made is that now `:path` for an `:unknown` event is `nil` instead of `maybe` `File`.

In the current implementation we know that `:unknown` will always have a `nil` `:path`. Alternatively, if we did something more like:
```clj
  :path (if-let [context (.context event)]
          (-> watch-path
            (.resolve context)
            fs/file)
          nil)
```
Then we wouldn't necessarily know when `:path` was `nil` and we could so something like this:
```clj
(def Event
  "Schema for an event on a file watched by this service."
  {:type (schema/enum :create :modify :delete :unknown)
   :path (schema/maybe File)})
```

The current impl in this PR is more brittle in the case that a normal event could return a null context, but I think the current impl more explicitly describes our (or at least my) current understanding of how the world works.